apps/external-contracts-tests/test-suites/national-registry.spec.ts

```diff
@@ -1,24 +1,55 @@
-// NOTE: To run this locally, you'll need to portforward soffia and set
-// the environment variable "SOFFIA_SOAP_URL" to https://localhost:8443
-// kubectl port-forward svc/socat-soffia 8443:443 -n socat
-import { NationalRegistryApi } from '@island.is/clients/national-registry-v1'
+// NOTE: To run this locally, you'll need to portforward xroad service port and set
+// the environment variable "XROAD_BASE_PATH_WITH_ENV=http://localhost:8088/r1/IS-TEST"
+// Example:
+//
+// $ kubectl port-forward svc/socat-xroad 8088:80 -n socat &
+// $ XROAD_BASE_PATH_WITH_ENV=http://localhost:8088/r1/IS-DEV XROAD_CLIENT_ID=IS-DEV/GOV/10000/island-is-client XROAD_TJODSKRA_API_PATH=/SKRA-Protected/Einstaklingar-v1 XROAD_TJODSKRA_MEMBER_CODE=10001 yarn nx run external-contracts-tests:external-test
+
+import { Test, TestingModule } from '@nestjs/testing'
+import {
+  createXRoadAPIPath,
+  XRoadMemberClass,
+} from '@island.is/shared/utils/server'
+
+import {
+  NationalRegistryModule,
+  EinstaklingarApi,
+} from '@island.is/clients/national-registry-v2'
 
 describe('is.island.external.national', () => {
-  let client: NationalRegistryApi
+  let client: EinstaklingarApi
+  const XROAD_CLIENT = process.env.XROAD_CLIENT_ID!
+
   beforeAll(async () => {
-    client = await NationalRegistryApi.instantiateClass({
-      baseSoapUrl: process.env.SOFFIA_SOAP_URL!,
-      user: process.env.SOFFIA_USER!,
-      password: process.env.SOFFIA_PASS!,
-      host: process.env.SOFFIA_HOST_URL!,
-    })
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        NationalRegistryModule.register({
+          xRoadPath: createXRoadAPIPath(
+            process.env.XROAD_BASE_PATH_WITH_ENV!,
+            XRoadMemberClass.GovernmentInstitution,
+            process.env.XROAD_TJODSKRA_MEMBER_CODE!,
+            process.env.XROAD_TJODSKRA_API_PATH!,
+          ),
+          xRoadClient: XROAD_CLIENT,
+        }),
+      ],
+    }).compile()
+    client = await module.get(EinstaklingarApi)
   })
   it('should get user correctly', async () => {
-    const user = await client.getUser('0101302989')
-    expect(user.Fulltnafn).toEqual('Gervimaður Ameríka')
+    const user = await client.einstaklingarGetEinstaklingur({
+      id: '0101302989',
+      xRoadClient: XROAD_CLIENT,
+    })
+    expect(user.fulltNafn).toEqual('Gervimaður Ameríka')
```

Comment:
Ahh ... this API needs an IDS access token with specific scopes. Need to set up a machine client and configure EnhancedFetch#autoAuth. Hopefully we can call their primary endpoint without a nationalId claim.

But there are some endpoints (in this service and others) which require a nationalId claim, so we should also discuss how we'll set up external-contract-tests against those. Eg with Gervimenn and some custom OAuth2 flow on dev/staging?